factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/registry/impl/IndexFileCalculatorImpl.kt

```diff
@@ -29,67 +30,82 @@ import org.factcast.schema.registry.cli.utils.mapEventTransformations
-                    event.type,
-                    transformation.from,
-                    transformation.to,
-                    checksumService.createMd5Hash(transformation.transformationPath)
-                )
-            }
-
-        val syntheticTransformations = project
-            .mapEvents { namespace, event ->
-                missingTransformationCalculator.calculateDowncastTransformations(event).map {
-                    val (fromVersion, toVersion) = it
+                .mapEventTransformations { namespace, event, transformation ->
                     val id = getTransformationId(
-                        namespace,
-                        event,
-                        fromVersion.version,
-                        toVersion.version
+                            namespace,
+                            event,
+                            transformation.from,
+                            transformation.to
                     )
 
-                    SyntheticTransformation(
-                        id,
-                        namespace.name,
-                        event.type,
-                        fromVersion.version,
-                        toVersion.version
+                    FileBasedTransformation(
+                            id,
+                            namespace.name,
+                            event.type,
+                            transformation.from,
+                            transformation.to,
+                            checksumService.createMd5Hash(transformation.transformationPath)
                     )
                 }
-            }.flatten()
+
+        val syntheticTransformations = project
+                .mapEvents { namespace, event ->
+                    missingTransformationCalculator.calculateDowncastTransformations(event).map {
+                        val (fromVersion, toVersion) = it
+                        val id = getTransformationId(
+                                namespace,
+                                event,
+                                fromVersion.version,
+                                toVersion.version
+                        )
+
+                        SyntheticTransformation(
+                                id,
+                                namespace.name,
+                                event.type,
+                                fromVersion.version,
+                                toVersion.version
+                        )
+                    }
+                }.flatten()
 
         return Index(schemas, transformations.plus(syntheticTransformations))
     }
+
+    private fun createMd5Hash(filePath: Path, schemaStripTitles: Boolean): String =
+            if (schemaStripTitles) createTitleFilteredMd5Hash(filePath)
+                else checksumService.createMd5Hash(filePath)
+
+    private fun createTitleFilteredMd5Hash(filePath: Path): String {
+        val filteredJsonNode = titleFilterService.filter(
```

Comment:
using a "service" here is probably overblown as it must not have state, and does not have dependencies either.
a simple function would do here.

Reply:
Also, TitleFilterServiceImpl deals with nullable, which it should not?

Reply:
> using a "service" here is probably overblown as it must not have state, and does not have dependencies either.
> a simple function would do here.

I totally agree. For some reasons we've implemented it as a filter in the first place. You're right - sharing a stateless function can be done much easier.

Reply:
@samba2 do you want to commit the nullability issue?